core/src/test/java/io/questdb/test/griffin/engine/window/WindowFunctionTest.java

```diff
@@ -4079,18 +4590,447 @@ public void testWindowFunctionDoesSortIfOrderByIsNotCompatibleWithBaseQuery() th
         });
     }
 
+    @Test
+    public void testLagLeadOverPartitionBy() throws Exception {
+
+        assertMemoryLeak(() -> {
+            execute("create table tab (ts timestamp, i long, j long, d double, s symbol, c VARCHAR) timestamp(ts)");
+            execute("insert into tab select x::timestamp, x/4, case when x % 3 = 0 THEN NULL ELSE x%5 END, x%5, 'k' || (x%5) ::symbol, 'k' || x from long_sequence(7)");
+
+            assertQueryNoLeakCheck(
+                    "ts\ti\tj\td\tlead\tlag\tlead_ignore_nulls\tlag_ignore_nulls\tlead1\tlag1\tlead2\tlag2\n" +
+                            "1970-01-01T00:00:00.000001Z\t0\t1\t1.0\t2.0\tnull\t2.0\tnull\t2.0\tnull\t2.0\tnull\n" +
+                            "1970-01-01T00:00:00.000002Z\t0\t2\t2.0\tnull\t1.0\tnull\t1.0\tnull\t1.0\t3.0\t1.0\n" +
+                            "1970-01-01T00:00:00.000003Z\t0\tnull\t3.0\tnull\t2.0\tnull\t2.0\tnull\t2.0\tnull\t2.0\n" +
+                            "1970-01-01T00:00:00.000004Z\t1\t4\t4.0\t0.0\tnull\t0.0\tnull\t0.0\tnull\t0.0\tnull\n" +
+                            "1970-01-01T00:00:00.000005Z\t1\t0\t0.0\tnull\t4.0\t2.0\t4.0\tnull\t4.0\t1.0\t4.0\n" +
+                            "1970-01-01T00:00:00.000006Z\t1\tnull\t1.0\t2.0\t0.0\t2.0\t0.0\t2.0\t0.0\t2.0\t0.0\n" +
+                            "1970-01-01T00:00:00.000007Z\t1\t2\t2.0\tnull\tnull\tnull\t0.0\tnull\tnull\tnull\t1.0\n",
+                    "select ts, i, j, d, " +
+                            "lead(j) over (partition by i), " +
```

Comment:
testing `LEAD` and `LAG` together forces execution to use CachedFactory. This factory uses a different codepath to exercise pass1. As a result, LagFunction.getDouble() is not invoked in the test. It would be good to also test `LAG` in SQL without `LEAD`